core/config/v2/types.go

```diff
@@ -1002,3 +1003,32 @@ func (ins *Insecure) setFrom(f *Insecure) {
 		ins.OCRDevelopmentMode = f.OCRDevelopmentMode
 	}
 }
+
+type MercuryCredentials struct {
+	Name     *string
+	URL      *models.SecretURL
+	Username *models.Secret
+	Password *models.Secret
+}
+
+type MercurySecrets struct {
+	Credentials []MercuryCredentials
+}
```

Comment:
If you want to try a map, I think this would work:
```suggestion
type MercuryCredentials struct {
	URL      *models.SecretURL
	Username *models.Secret
	Password *models.Secret
}

type MercurySecrets struct {
	Credentials map[string]MercuryCredentials
}
```
And then you specify it like:
```toml
[Mercury.Credentials.cred1]
URL = "https://chain1.link"
Username = "username1"
Password = "password1"

[Mercury.Credentials.cred2]
URL = "https://chain2.link"
Username = "username2"
Password = "password2"

```